ddl/partition.go

```diff
@@ -481,6 +488,535 @@ func buildTablePartitionInfo(ctx sessionctx.Context, s *ast.PartitionOptions, tb
 	return nil
 }
 
+func isPartExprUnsigned(ctx sessionctx.Context, tbInfo *model.TableInfo) bool {
+	expr, err := expression.ParseSimpleExprWithTableInfo(ctx, tbInfo.Partition.Expr, tbInfo)
+	if err != nil {
+		return false
+	}
+	pCols := expression.ExtractColumns(expr)
+	for _, col := range pCols {
+		if mysql.HasUnsignedFlag(col.GetType().GetFlag()) {
+			return true
+		}
+	}
+	return false
+}
+
+// getPartitionIntervalFromTable checks if a partitioned table matches a generated INTERVAL partitioned scheme
+func getPartitionIntervalFromTable(ctx sessionctx.Context, tbInfo *model.TableInfo) *ast.PartitionInterval {
+	if tbInfo.Partition == nil ||
+		tbInfo.Partition.Type != model.PartitionTypeRange {
+		return nil
+	}
+	if len(tbInfo.Partition.Columns) > 1 {
+		// Multi-column RANGE COLUMNS is not supported with INTERVAL
+		return nil
+	}
+	if len(tbInfo.Partition.Definitions) < 2 {
+		return nil
+	}
+
+	var (
+		interval ast.PartitionInterval
+		startIdx int            = 0
+		endIdx   int            = len(tbInfo.Partition.Definitions) - 1
+		colType  types.EvalType = types.ETInt
+		minVal   string         = "0"
+	)
+	if len(tbInfo.Partition.Columns) > 0 {
+		partCol := findColumnByName(tbInfo.Partition.Columns[0].L, tbInfo)
+		if partCol.FieldType.EvalType() == types.ETInt {
+			min := getLowerBoundInt(partCol)
+			minVal = strconv.FormatInt(min, 10)
+		} else if partCol.FieldType.EvalType() == types.ETDatetime {
+			minVal = "0000-01-01"
+		} else {
+			return nil
```

Comment:
I hope there is a comment for every invalid case.

Reply:
I added some comments and restructured the code a little bit in this function, as well as added a function comment that `nil` will be return in case of error.